base/shell.jl

```diff
@@ -252,60 +252,103 @@ shell_escape_posixly(args::AbstractString...) =
 end
-
+shell_escape_wincmd(s::AbstractString) = sprint(shell_escape_wincmd, s;
+                                                sizehint = 2*sizeof(s))
 
 """
-     shell_escaped_winsomely(args::Union{Cmd,AbstractString...})::String
-
-Convert the collection of strings `args` into single string suitable for passing as the argument
-string for a Windows command line. Windows passes the entire command line as a single string to
-the application (unlike POSIX systems, where the list of arguments are passed separately).
-Many Windows API applications (including julia.exe), use the conventions of the [Microsoft C
-runtime](https://docs.microsoft.com/en-us/cpp/c-language/parsing-c-command-line-arguments) to
-split that command line into a list of strings. This function implements the inverse of such a
-C runtime command-line parser. It joins command-line arguments to be passed to a Windows console
-application into a command line, escaping or quoting meta characters such as space,
-double quotes and backslash where needed. This may be useful in concert with the `windows_verbatim`
-flag to [`Cmd`](@ref) when constructing process pipelines.
+    escape_microsoft_c_args(args::Union{Cmd,AbstractString...})
+    escape_microsoft_c_args(io::IO, args::Union{Cmd,AbstractString...})
 
-# Example
-```jldoctest
-julia> println(shell_escaped_winsomely("A B\\", "C"))
-"A B\\" C
+Convert a collection of string arguments into a string that can be
+passed to many Windows command-line applications.
+
+Microsoft Windows passes the entire command line as a single string to
+the application (unlike POSIX systems, where the shell splits the
+command line into a list of arguments). Many Windows API applications
+(including julia.exe), use the conventions of the [Microsoft C/C++
+runtime](https://docs.microsoft.com/en-us/cpp/c-language/parsing-c-command-line-arguments)
+to split that command line into a list of strings.
+
+This function implements an inverse for a parser compatible with these rules.
+It joins command-line arguments to be passed to a Windows
+C/C++/Julia application into a command line, escaping or quoting the
+meta characters space, TAB, double quote and backslash where needed.
+
+See also: [`shell_escape_wincmd`](@ref), [`escape_raw_string`](@ref)
 """
-shell_escape_winsomely(args::AbstractString...) =
-    sprint(print_shell_escaped_winsomely, args..., sizehint=(sum(length, args)) + 3*length(args))
+function escape_microsoft_c_args(io::IO, args::AbstractString...)
+    # http://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULES
+    first = true
+    for arg in args
+        if first
+            first = false
+        else
+            write(io, ' ')  # separator
+        end
+        if isempty(arg) || occursin(r"[ \t\"]", arg)
+            # Julia raw strings happen to use the same escaping convention
+            # as the argv[] parser in Microsoft's C runtime library.
+            write(io, '"')
+            escape_raw_string(io, arg)
```

Comment:
Did you intend not to wrap a single argument in quotes? 

E.g.: ```Base.escape_microsoft_c_args("a") == "a" and not  "\"a\"" ```

I'm aware technically it doesn't matter, but since we are doing it when the number of arguments is greater than one it doesn't quite seem consistent, albeit I don't have a strong preference either way.

Reply:
> Did you intend not to wrap a single argument in quotes?

Please look again. This function wraps an argument only in quotes if that is necessary for that argument (i.e., if the argument is the empty string or contains space, TAB or "). The decision whether to wrap an argument does _not_ depend on the number of arguments. (Why do you think it does?)

In the interest of the readability of the output (which may show up in error messages, etc.), this function adds additional meta-characters only where necessary, independent of the number of arguments. Anything else does not need to be escaped for the purpose of being digested by the ARGV parser in the Microsoft C runtime.